certification/internal/shell/mounted_engine.go

```diff
@@ -0,0 +1,63 @@
+package shell
+
+import (
+	"time"
+
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification"
+	containerutil "github.com/redhat-openshift-ecosystem/openshift-preflight/certification/internal/utils/container"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/runtime"
+	log "github.com/sirupsen/logrus"
+)
+
+type MountedCheckEngine struct {
+	Image string
+	Check certification.Check
+
+	results runtime.Results
+}
+
+// ExecuteChecks runs all checks stored in the check engine.
+func (e *MountedCheckEngine) ExecuteChecks() error {
+	log.Info("target image: ", e.Image)
+	// check if the image needs downloading
```

Comment:
can probably remove this comment as we're assuming the image has been downloaded before we ever reach this point.

Reply:
Done.